discord/app_commands/tree.py

```diff
@@ -511,29 +511,44 @@ def get_commands(
         type: AppCommandType,
     ) -> Union[List[Union[Command[Any, ..., Any], Group]], List[ContextMenu]]:
         ...
+        
+    @overload
+    def get_commands(
+        self,
+        *,
+        guild: Optional[Snowflake] = ...,
+        type: Optional[AppCommandType] = ...,
+    ) -> List[Union[Command[Any, ..., Any], Group, ContextMenu]]:
+        ...
 
     def get_commands(
         self,
         *,
         guild: Optional[Snowflake] = None,
-        type: AppCommandType = AppCommandType.chat_input,
-    ) -> Union[List[Union[Command[Any, ..., Any], Group]], List[ContextMenu]]:
+        type: Optional[AppCommandType] = None,
+    ) -> Union[
+        List[ContextMenu],
+        List[Union[Command[Any, ..., Any], Group]], 
+        List[Union[Command[Any, ..., Any], Group, ContextMenu]],
+    ]:
         """Gets all application commands from the tree.
 
         Parameters
         -----------
         guild: Optional[:class:`~discord.abc.Snowflake`]
             The guild to get the commands from, not including global commands.
             If not given or ``None`` then only global commands are returned.
-        type: :class:`~discord.AppCommandType`
-            The type of commands to get. Defaults to :attr:`~discord.AppCommandType.chat_input`,
-            i.e. slash commands.
+        type: Optional[:class:`~discord.AppCommandType`]
+            The type of commands to get. When not given or ``None``, then all
+            the commands for the given guild are returned.
 
         Returns
         ---------
-        Union[List[:class:`ContextMenu`], List[Union[:class:`Command`, :class:`Group`]]
+        Union[List[:class:`ContextMenu`], List[Union[:class:`Command`, :class:`Group`], List[:class:`Command`, :class:`Group`, :class:`ContextMenu`]]
```

Comment:
```suggestion
        List[Union[:class:`ContextMenu`, :class:`Command`, :class:`Group`]]
```

Reply:
I was thinking of implementing it like this but was unsure due to the already existing overwrites 😊 - thank you